src/main/protobuf/soc/gameState.proto

```diff
@@ -0,0 +1,19 @@
+syntax = "proto2";
+
+package protos.soc;
+
+import "soc/board.proto";
+import "soc/inventory.proto";
+import "soc/playerState.proto";
+
+message GameState {
+    required Board board = 1;
+    repeated PlayerState playerStates = 2;
+    required int32 developmentCardsInDeck = 3;
+    required int32 currentTurn = 4;
+    required bool canRollDice = 5;
```

Comment:
Do these booleans provide information other than just if a player has rolled or not? 

ie, could we just have a single boolean `isPreRoll` .... I don't suggest that, but I'm trying to understand the nature of these flags. 

I would probably prefer something like a nested `TurnPhase` enum.

Reply:
So I thought about a `TurnPhase` enum but the problems comes with `can_play_development_card`. Would that info be nested into the  `TurnPhase`? That would result in 4 phases `RollOrCard`, `Roll`, `BuildTradeCardOrEnd`, and `BuildTradeOrEnd`. And then we need logic to go between them.

Reply:
If we associate palyed development cards, with the turn in which they were played, then we can just have a `TurnPhase` with two types `ROLL, `MAIN` or `ROLL` and `PLAY` or whatever we want to call the two phases.

Reply:
I do not understand this